### PR TITLE
Various bugfixes / enhancements

### PR DESCRIPTION
**Pull request by silentwings**
re pm

### CODE REVIEW

**Comment by silentwings** [comment]
This is ready now.

**Comment by abma** [comment]
flag_stats is not used anywhere?

File: protocol/Channel.py [hunk 1]
```diff
@@ -28,8 +27,8 @@ def broadcast(self, message, ignore=set([])):
 	def channelMessage(self, message):
 		self.broadcast('CHANNELMESSAGE %s %s' % (self.name, message))
 
-	def register(self, client, owner_user_id):
-		self.owner_user_id = owner.db_id
+	def register(self, client, owner_user_id): # fixme: unused?
```
Comment (abma) [inline protocol/Channel.py]: when unused: remove it? :)

File: protocol/Protocol.py [hunk 2]
```diff
@@ -934,6 +952,14 @@ def in_LOGIN(self, client, username, password='', cpu='0', local_ip='', sentence
...
 					self.out_SERVERMSG(client, 'Unsupported/unknown compatibility flag(s) in LOGIN: %s' % (unsupported), True)
+					
+				# record flag stats
+				self.flag_stats['n_samples'] += 1
+				for flag in flags:
+					if flag in self.flag_stats:
+						self.flag_stats[flag] += 1
+					else:
+						self.flag_stats[flag] = 1					
 			try:
```
Comment (abma) [inline protocol/Protocol.py]: flag_stats seems to be used nowhere

File: protocol/Channel.py (hunk 1 above)
Comment (silentwings) [reply protocol/Channel.py]: It is unused here - but it is used in the other PR! So better to leave it in here, I think.

File: protocol/Protocol.py (hunk 2 above)
Comment (silentwings) [reply protocol/Protocol.py]: Yes, see https://github.com/spring/uberserver/issues/270#issuecomment-432333604

Afaics the stat collection that already existed (for cmd counts) didn't actually output to anywhere. Without seeing inside the live server I don't know what the best place/way for it to output stats is. I could add a server cmd to just print in plain-text, what's your preference?

Comment (abma) [reply protocol/Protocol.py]: its outputed to the log file on reload...

Comment (silentwings) [reply protocol/Protocol.py]: Ah, missed that - I've added it for flag_stats too. Having to call a RELOAD just to print out stats seems a little strange to me.

**Comment by abma** [comment]
i guess the "review part" won't get better: needs to be tested in productive.

**Pull request merged**